src/Types/BaseType.hack

```diff
@@ -1,13 +1,32 @@
 namespace Slack\GraphQL\Types;
 
-// TODO: Implement introspection: https://spec.graphql.org/draft/#sec-Introspection
-// It may or may not be a good idea to reuse the existing codegen to generate the introspection types.
-// If we did that, then this would be:
-// <<GraphQL\Object('__Type', '')>>
-abstract class BaseType {
-    // <<GraphQL\Field('name', '')>>
+use namespace Slack\GraphQL;
+
+abstract class BaseType implements GraphQL\Introspection\__Type {
```

Comment:
If you try my suggestion from above (have separate `getType()` and `getIntrospectionType()` methods on field definition etc.) you'd probably want to remove the `implements GraphQL\Introspection\__Type` and add it to all subtypes except `Nullable...` -- that would ensure that we don't return an invalid kind of wrapper at runtime/during introspection

Actually it might be even better if we put `implements GraphQL\Introspection\__Type` only on 1 copy of each type that has input+output version, that would prevent us from e.g. returning duplicate types from `$schema->getTypes()`

Reply:
I like this, let me try it out